### PR TITLE
fixes #1951 - support for libvirt named networks

### PR DESCRIPTION
**Pull request by domcleal**

### CODE REVIEW

File: app/views/compute_resources_vms/form/libvirt/_network.html.erb [hunk 1]
```diff
@@ -1,9 +1,22 @@
...
-  <% if (networks = compute_resource.networks).any? -%>,
-    <%= selectable_f f, :bridge, networks.map(&:name), { }, :class => "span2", :label => "Network",
-                     :help_inline => remove_child_link("X", f, { :method => :'_delete', :title => 'remove network interface', :class => 'label label-important' }) %>
-  <% else -%>
-    <%= text_f f, :bridge, :class => "span2", :label => "Network", :help_block => "your libvirt host does not support interface listing, please enter here the bridge name (e.g. br0)",
-               :help_inline => remove_child_link("X", f, { :method => :'_delete', :title => 'remove network interface', :class => 'label label-important' }) %>
+  <%= field(f, :type, :label => 'Network Type', :help_inline => remove_child_link("X", f, { :method => :'_delete', :title => 'remove network interface', :class => 'label label-important' })) do -%>
+    <div>
+      <% networks = compute_resource.networks -%>
+      <%= radio_button_f f, :type, {:value=>'network', :checked => networks.any?, :onclick => 'libvirt_network_type(this)', :text => 'Virtual Network'} -%>
```
Comment (ohadlevy) [inline app/views/compute_resources_vms/form/libvirt/_network.html.erb]: what does the JS do? its not part of the commit?

Comment (domcleal) [inline app/views/compute_resources_vms/form/libvirt/_network.html.erb]: oops, I'd meant to remove that.  I was going to disable the field not in use, but it was a bit fiddly.

**Comment by domcleal** [comment]
Branch updated

File: app/views/compute_resources_vms/form/libvirt/_network.html.erb [hunk 2]
```diff
@@ -1,9 +1,22 @@
 <div class="fields">
-  <% if (networks = compute_resource.networks).any? -%>,
-    <%= selectable_f f, :bridge, networks.map(&:name), { }, :class => "span2", :label => "Network",
-                     :help_inline => remove_child_link("X", f, { :method => :'_delete', :title => 'remove network interface', :class => 'label label-important' }) %>
-  <% else -%>
-    <%= text_f f, :bridge, :class => "span2", :label => "Network", :help_block => "your libvirt host does not support interface listing, please enter here the bridge name (e.g. br0)",
-               :help_inline => remove_child_link("X", f, { :method => :'_delete', :title => 'remove network interface', :class => 'label label-important' }) %>
+  <%= field(f, :type, :label => 'Network Type', :help_inline => remove_child_link("X", f, { :method => :'_delete', :title => 'remove network interface', :class => 'label label-important' })) do -%>
```
Comment (ohadlevy) [inline app/views/compute_resources_vms/form/libvirt/_network.html.erb:2]: NOT a big deal but:

maybe this is better done in JS instead, e.g. if you have a radio button and one select (that does the show & enable or hide & disable).

its a UI issue, but seeing the two dropdowns makes the form a bit more complicated, esp when you have  many interfaces.

Comment (ohadlevy) [inline app/views/compute_resources_vms/form/libvirt/_network.html.erb:2]: @abenari any feedback?

**Comment by ohadlevy** [comment]
@domcleal  any chance you could fix it so it looks correct for /compute_resources/id/vms/new too ?

File: app/views/compute_resources_vms/form/libvirt/_network.html.erb (hunk 2 above)
Comment (domcleal) [inline app/views/compute_resources_vms/form/libvirt/_network.html.erb:2]: At the moment I've got:

```
o Virtual Network
    Name: [ list of networks ]
o Bridge
    Interface: [ list of bridges ]
```

Would you suggest just hiding the select that isn't in use (so it appears to fold each section)?  Or I could have just one select for both types and drop it down below the radio buttons.

Comment (ohadlevy) [inline app/views/compute_resources_vms/form/libvirt/_network.html.erb:2]: I was thinking on one visible select, that changes values based on the
radio?

**Comment by domcleal** [comment]
Closing until I get around to the above changes.
